ocaml/xapi-types/ref.ml

```diff
@@ -32,28 +31,41 @@ let of_string x = x
 (* embedded into the reference .... *)
 
 (* a dummy reference is a reference of an object which is not in database *)
-let dummy_sep = '|'
+let dummy_sep = "|"
 let dummy_prefix = "DummyRef:"
 
-open Xstringext
-
 let make_dummy task_name =
-  let uuid = Uuid.string_of_uuid (Uuid.make_uuid ()) in
-  dummy_prefix ^ String.of_char dummy_sep ^ uuid ^ String.of_char dummy_sep ^ task_name
+  let uuid = Uuidm.v `V4 |> Uuidm.to_string in
+  dummy_prefix ^ dummy_sep ^ uuid ^ dummy_sep ^ task_name
 
 let is_dummy x =
-  String.startswith dummy_prefix x
+  Astring.String.is_prefix ~affix:dummy_prefix x
+
+let cut_in_three ~sep str =
+  let (>>=) opt f =
+    match opt with
+    | Some value -> f value
+    | None -> None
+  in
+  let open Astring in
+  String.cut ~sep str
+  >>= fun (head, tail) ->
+  String.cut ~sep tail
+  >>= fun (head', tail) ->
+  Some (head, head', tail)
 
 let name_of_dummy x =
-  match String.split ~limit:3 dummy_sep x with
-  | [_;_;name] -> name
-  | l -> failwith (Printf.sprintf "Ref.name_of_dummy: %s is not a valid dummy reference (%i)" x (List.length l))
+  match cut_in_three ~sep:dummy_sep x with
+  | Some(_, _, name) -> name
+  | None -> failwith (
+      Printf.sprintf "Ref.name_of_dummy: %s is not a valid dummy reference" x)
 
 (* we do not show the name when we pretty print the dummy reference *)
 let pretty_string_of_dummy x =
-  match String.split ~limit:3 dummy_sep x with
```

Comment:
Is there no `split` function in `Astring`?

Reply:
Not with an equivalent of `~limit:3`, and it's pointless to split to recombine later when we can use the cut twice in a safe way